python/spec/dependabot/python/file_updater/poetry_file_updater_spec.rb

```diff
@@ -501,6 +501,137 @@
         end
       end
 
+      context "with inline comments in the dependencies groups" do
+        let(:pyproject_fixture_name) { "inline_comments.toml" }
+        let(:dependency) do
+          Dependabot::Dependency.new(
+            name: dependency_name,
+            version: "1.27.2",
+            previous_version: "1.18.1",
+            package_manager: "pip",
+            requirements: [{
+              requirement: "^4.19.1",
+              file: "pyproject.toml",
+              source: nil,
+              groups: ["dependencies"]
+            }, {
+              requirement: "^5.12.0",
+              file: "pyproject.toml",
+              source: nil,
+              groups: ["dev"]
+            }, {
+              requirement: ">=3.12.0",
+              file: "pyproject.toml",
+              source: nil,
+              groups: ["test"]
+            }],
+            previous_requirements: [{
+              requirement: "^4.18.5",
+              file: "pyproject.toml",
+              source: nil,
+              groups: ["dependencies"]
+            }, {
+              requirement: "^5.9.3",
+              file: "pyproject.toml",
+              source: nil,
+              groups: ["dev"]
+            }, {
+              requirement: ">=3.8.2",
+              file: "pyproject.toml",
+              source: nil,
+              groups: ["test"]
+            }]
+          )
```

Comment:
In real life, all `requirements` here will be for the same dependency, so to better represent real behavior, each dependency should configure the requirements array for the dependency it's testing (similarly to how you customized `dependency_name`. Does that make sense? Basically, I think you'd need to add a `let(:requirements)` to each spec, and use that here.  Does that make sense?

Reply:
Certainly, I agree with your suggestion.
Thanks for the review, done 👍